src/Illuminate/Cache/Repository.php

```diff
@@ -262,12 +264,14 @@ public function setMultiple($values, $ttl = null)
      *
      * @param  string  $key
      * @param  mixed   $value
-     * @param  \DateTimeInterface|\DateInterval|float|int  $minutes
+     * @param  \DateTimeInterface|\DateInterval|float|int|null  $minutes
      * @return bool
      */
     public function add($key, $value, $minutes)
     {
-        if (is_null($minutes = $this->getMinutes($minutes))) {
+        $minutes = $this->getMinutes($minutes);
+
+        if ($minutes !== null && $minutes <= 0) {
```

Comment:
why is this `$minutes !== null`? Shouldnt we be returning `false` if `$minutes` is `null`?

Reply:
That's what this patch is fixing: the driver should still be hit, if `null` is given. The added tests do highlight that.